src/modules/processing/services/tiktok.js

```diff
@@ -19,54 +19,54 @@ export default async function(obj) {
 
         if (!html) return { error: 'ErrorCouldntFetch' };
 
-        if (html.slice(0, 17) === '<a href="https://') {
-            postId = html.split('<a href="https://')[1].split('?')[0].split('/')[3]
-        } else if (html.slice(0, 32) === '<a href="https://m.tiktok.com/v/' && html.includes('/v/')) {
-            postId = html.split('/v/')[1].split('.html')[0].replace("/", '')
+        if (html.startsWith('<a href="https://')) {
+            const { patternMatch } = extract(html.split('<a href="https://')[1].split('?')[0])
+            postId = patternMatch.postId
         }
     }
     if (!postId) return { error: 'ErrorCantGetID' };
 
-    let deviceInfo = new URLSearchParams(env.tiktokDeviceInfo).toString();
-
-    let apiURL = new URL(apiPath);
-        apiURL.searchParams.append("aweme_id", postId);
-
-    let detail = await fetch(`${apiURL.href}&${deviceInfo}`, {
+    // should always be /video/, even for photos
+    const res = await fetch(`https://tiktok.com/@i/video/${postId}`, {
         headers: {
-            "user-agent": apiUserAgent
+            "user-agent": genericUserAgent,
+            cookie,
         }
-    }).then(r => r.json()).catch(() => {});
+    })
+    updateCookie(cookie, res.headers)
+
+    const html = await res.text()
 
-    detail = detail?.aweme_list?.find(v => v.aweme_id === postId);
-    if (!detail) return { error: 'ErrorCouldntFetch' };
+    let detail
+    try {
+        const json = html
+            .split('<script id="__UNIVERSAL_DATA_FOR_REHYDRATION__" type="application/json">')[1]
+            .split('</script>')[0]
+        const data = JSON.parse(json)
+        detail = data["__DEFAULT_SCOPE__"]["webapp.video-detail"]["itemInfo"]["itemStruct"]
+    } catch {
+        return { error: 'ErrorCouldntFetch' };
+    }
 
     let video, videoFilename, audioFilename, audio, images,
-        filenameBase = `tiktok_${detail.author.unique_id}_${postId}`,
+        filenameBase = `tiktok_${detail.author.uniqueId}_${postId}`,
         bestAudio = 'm4a';
 
-    images = detail.image_post_info?.images;
+    images = detail.imagePost?.images;
 
-    let playAddr = detail.video.play_addr_h264;
+    let playAddr = detail.video.playAddr;
     if (obj.h265) {
-        playAddr = detail.video.bit_rate[0].play_addr
-    }
-    if (!playAddr && detail.video.play_addr) {
-        playAddr = detail.video.play_addr
+        const h265PlayAddr = detail.video.bitrateInfo.find(b => b.CodecType.includes("h265"))?.PlayAddr.UrlList[0]
+        playAddr = h265PlayAddr || playAddr
     }
 
     if (!obj.isAudioOnly && !images) {
-        video = playAddr.url_list[0];
+        video = playAddr;
         videoFilename = `${filenameBase}.mp4`;
     } else {
-        let fallback = playAddr.url_list[0];
-        audio = fallback;
+        audio = detail.music.playUrl || playAddr;
         audioFilename = `${filenameBase}_audio`;
-        if (obj.fullAudio || fallback.includes("music")) {
```

Comment:
please add back full audio functionality! use old code as base, but update variables to match new json format

Reply:
not sure why it was removed at all, probably on accident

Reply:
default should be audio extraction from video, not full original audio